app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/welcome/WelcomeOnBoardScreen.kt

```diff
@@ -13,79 +14,125 @@ import androidx.compose.material3.MaterialTheme
-            textStyle =
-                poppinsTypography.headlineMedium.copy(
-                    fontSize = 16.sp, fontWeight = FontWeight.SemiBold),
-            modifier =
-                Modifier.weight(0.5f).semantics {
-                  testTag = C.Tag.welcomeOnBoardScreenSwitchWorkerButton
-                })
+    }
+    if (isWaitingForWorkerMode) {
+      Column(
+          modifier = Modifier.fillMaxSize(),
+          horizontalAlignment = Alignment.CenterHorizontally,
+          verticalArrangement = Arrangement.Center) {
+            androidx.compose.material3.CircularProgressIndicator(
+                color = colorScheme.primary, modifier = Modifier.testTag("Loader"))
+          }
+    } else {
+      Column(
+          modifier = Modifier.fillMaxSize().padding(16.dp),
+          horizontalAlignment = Alignment.CenterHorizontally,
+      ) {
+        Spacer(modifier = Modifier.height(100.dp * heightRatio.value))
+        Text(
+            "Welcome on board !!",
+            style =
+                poppinsTypography.headlineLarge.copy(
+                    color = MaterialTheme.colorScheme.onBackground, fontSize = 26.sp))
+        Spacer(modifier = Modifier.weight(0.2f))
+        Row(modifier = Modifier.weight(0.9f)) {
+          Image(
+              painter = painterResource(id = R.drawable.onboarding_worker_1),
+              contentDescription = "Description of the image",
+              contentScale = ContentScale.Crop,
+              modifier =
+                  Modifier.fillMaxSize().semantics { testTag = C.Tag.welcomeOnBoardScreenImage })
+        }
+        Spacer(modifier = Modifier.weight(0.2f))
+
+        // Show a loader while waiting for isWorker to become true
+
+        Row(modifier = Modifier.fillMaxWidth().weight(0.8f)) {
+          // "Stay User Mode" button: If isWorker is already false, we can navigate immediately
+          // or just navigate directly without waiting since we don't need isWorker to change
+          QuickFixButton(
+              buttonText = "Stay User Mode",
+              onClickAction = {
+                if (!testingFlag) {
+                  isWaitingForWorkerMode = true
+                } else {
+                  navigationActions.navigateTo(UserScreen.PROFILE)
+                }
+              },
+              buttonColor = colorScheme.primary,
+              textColor = colorScheme.onPrimary,
+              textStyle =
+                  poppinsTypography.headlineMedium.copy(
+                      fontSize = 16.sp, fontWeight = FontWeight.SemiBold),
+              modifier =
+                  Modifier.weight(0.5f).semantics {
+                    testTag = C.Tag.welcomeOnBoardScreenStayUserButton
+                  })
+
+          // "Switch Worker Mode": We wait for isWorker to become true
+          QuickFixButton(
+              buttonText = "Switch Worker Mode",
+              onClickAction = {
+                if (!testingFlag) {
```

Comment:
Should probably remove this, I don't know why is it still here, could you add some documentation maybe?